homeassistant/components/sensor/dsmr.py

```diff
@@ -0,0 +1,153 @@
+"""
+Support for Dutch Smart Meter Requirements.
+
+Also known as: Smartmeter or P1 port.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.dsmr/
+"""
+
+import asyncio
+import logging
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_DEVICE
+from homeassistant.helpers.entity import Entity
+
+DOMAIN = 'dsmr'
+
+REQUIREMENTS = ['dsmr-parser==0.2']
+
+CONF_DSMR_VERSION = 'dsmr_version'
+DEFAULT_DEVICE = '/dev/ttyUSB0'
+DEFAULT_DSMR_VERSION = '4'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_DEVICE, default=DEFAULT_DEVICE): cv.string,
```

Comment:
Is there a reason it is not outlined in a document yet? 

In my opinion if you want to work toward a convention for (for example) naming you outline it first in a document, this should make it easy for (new) contributors to learn about what the conventions are. Now I had to learn this convention through a pullrequest review, as the only other information I could go on is other components.